src/test/java/bio/terra/service/tabulardata/google/BigQueryPdaoTest.java

```diff
@@ -288,12 +216,59 @@ public void datasetTest() throws Exception {
                 connectedOperations.handleCreateSnapshotSuccessCase(snapshotResponse);
             SnapshotModel snapshot = connectedOperations.getSnapshot(snapshotSummary.getId());
 
-            // TODO: Assert that the snapshot contains the rows we expect.
-            // Skipping that for now because there's no REST API to query table contents.
+            BigQueryProject bigQueryProject = bigQueryProjectForDatasetName(dataset.getName());
             Assert.assertThat(snapshot.getTables().size(), is(equalTo(3)));
+            List<String> participantIds = queryForIds(snapshot.getName(), "participant", bigQueryProject);
```

Comment:
I wonder if this section could also benefit from the type of deduplication done in the earlier test.

Reply:
It probably could. I tried `forEach` in one spot but it complained about `queryForIds` throwing a checked exception.